### PR TITLE
Update storage blob SDK from t1 to t2

### PR DESCRIPTION
**Pull request by zzkaii**
1. Update storage blob SDK from t1 to t2
2. Update files:

- soft-delete/App.config 
- soft-delete/Program.cs
- soft-delete/packages.config
- soft-delete/soft-delete.csproj

@scottaddie,@AlexGhiondea,@jongio,@kyle-patterson and @tg-msft for notification .

### CODE REVIEW

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@jongio,@kyle-patterson and @tg-msft   Could you please review this PR?

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@jongio,@kyle-patterson and @tg-msft Could you please review this PR?

File: soft-delete/Program.cs [hunk 1]
```diff
@@ -145,28 +138,29 @@ private static async Task SoftDeleteTest()
...
             try
             {
-                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(CloudConfigurationManager.GetSetting("StorageConnectionString"));
-                CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-                IRetryPolicy exponentialRetryPolicy = new ExponentialRetry(TimeSpan.FromSeconds(2), 10);
-                blobClient.DefaultRequestOptions.RetryPolicy = exponentialRetryPolicy;
+                BlobClientOptions blobClientOptions = new BlobClientOptions();
+                blobClientOptions.Retry.Mode = RetryMode.Exponential;
+                blobClientOptions.Retry.Delay = TimeSpan.FromSeconds(2);
+                blobClientOptions.Retry.MaxRetries = 10;
```
Comment (christothes) [inline soft-delete/Program.cs]: Just curious why these specific retry settings are important to the sample. 

As a side note, Exponential is already the default Mode, so that can be removed. In addition with exponential retry and a maxRetries of 10, I wonder if we need to set the initial delay so high - maybe that should also be the default.

File: soft-delete/packages.config [hunk 2]
```diff
@@ -1,15 +1,27 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net462" />
+  <package id="Azure.Core" version="1.17.0" targetFramework="net462" />
+  <package id="Azure.Storage.Blobs" version="12.10.0-beta.2" targetFramework="net462" />
+  <package id="Azure.Storage.Common" version="12.9.0-beta.2" targetFramework="net462" />
+  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net462" />
```
Comment (christothes) [inline soft-delete/packages.config]: I think Common will be brought in automatically as a dependency. Is the explicit reference needed?
I think Azure.Core is also automatically referenced.

```suggestion
  <package id="Azure.Core" version="1.20.0" targetFramework="net462" />
  <package id="Azure.Storage.Blobs" version="12.10.0" targetFramework="net462" />
  <package id="Azure.Storage.Common" version="12.9.0-beta.2" targetFramework="net462" />
  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net462" />
```

File: soft-delete/Program.cs (hunk 1 above)
Comment (zzkaii) [reply soft-delete/Program.cs]: Thank you for your advice. I've modified it.

File: soft-delete/packages.config (hunk 2 above)
Comment (zzkaii) [reply soft-delete/packages.config]: Thank you for your advice. I've updated it.

File: soft-delete/Program.cs [hunk 3]
```diff
@@ -145,28 +138,28 @@ private static async Task SoftDeleteTest()
...
         {
             try
             {
-                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(CloudConfigurationManager.GetSetting("StorageConnectionString"));
-                CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-                IRetryPolicy exponentialRetryPolicy = new ExponentialRetry(TimeSpan.FromSeconds(2), 10);
-                blobClient.DefaultRequestOptions.RetryPolicy = exponentialRetryPolicy;
+                BlobClientOptions blobClientOptions = new BlobClientOptions();
+                blobClientOptions.Retry.Delay = TimeSpan.FromSeconds(1);
+                blobClientOptions.Retry.MaxRetries = 1;
```
Comment (christothes) [inline soft-delete/Program.cs]: Should we just remove these? Also now the comment above should be updated to match.

File: soft-delete/packages.config [hunk 4]
```diff
@@ -1,15 +1,27 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net462" />
+  <package id="Azure.Core" version="1.20.0" targetFramework="net462" />
```
Comment (christothes) [inline soft-delete/packages.config:3]: remove?

File: soft-delete/packages.config [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
+  <package id="Azure.Storage.Blobs" version="12.10.0" targetFramework="net462" />
+  <package id="Azure.Storage.Common" version="12.9.0-beta.2" targetFramework="net462" />
```
Comment (christothes) [inline soft-delete/packages.config]: remove?

File: soft-delete/soft-delete.csproj [hunk 6]
```diff
@@ -35,9 +35,17 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Microsoft.Azure.KeyVault.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\packages\Microsoft.Azure.KeyVault.Core.2.0.4\lib\net45\Microsoft.Azure.KeyVault.Core.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Azure.Core, Version=1.17.0.0, Culture=neutral, PublicKeyToken=92742159e12e44c8, processorArchitecture=MSIL">
```
Comment (christothes) [inline soft-delete/soft-delete.csproj]: please run `dotnet restore` and `dotnet build` after changing the package references so that these are updated.

File: soft-delete/Program.cs (hunk 3 above)
Comment (zzkaii) [reply soft-delete/Program.cs]: Done.

File: soft-delete/packages.config (hunk 5 above)
Comment (zzkaii) [reply soft-delete/packages.config]: Done.

File: soft-delete/soft-delete.csproj (hunk 6 above)
Comment (zzkaii) [reply soft-delete/soft-delete.csproj]: Done.

File: soft-delete/packages.config [hunk 7]
```diff
@@ -1,15 +1,24 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net462" />
+  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net462" />
```
Comment (christothes) [inline soft-delete/packages.config:6]: What happened to `Azure.Storage.Blobs` ?

Comment (zzkaii) [reply soft-delete/packages.config:6]: Done.

File: soft-delete/App.config [hunk 8]
```diff
@@ -28,6 +28,30 @@
...
+        <bindingRedirect oldVersion="0.0.0.0-4.0.5.1" newVersion="4.0.5.1" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="System.ValueTuple" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-4.0.3.0" newVersion="4.0.3.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Azure.Core" publicKeyToken="92742159e12e44c8" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.17.0.0" newVersion="1.17.0.0" />
+      </dependentAssembly>
```
Comment (christothes) [inline soft-delete/App.config]: Why are these needed?

File: soft-delete/packages.config [hunk 9]
```diff
@@ -1,15 +1,25 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net462" />
+  <package id="Azure.Storage.Blobs" version="12.10.0" targetFramework="net462" />
+  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net462" />
   <package id="Microsoft.Data.Edm" version="5.8.3" targetFramework="net462" />
```
Comment (christothes) [inline soft-delete/packages.config]: Many of these can be deleted, I think - I think this is all we need:

```
  <package id="Azure.Core" version="1.19.0" targetFramework="net462" />
  <package id="Azure.Storage.Blobs" version="12.10.0" targetFramework="net462" />
  <package id="Azure.Storage.Common" version="12.9.0" targetFramework="net462" />
  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net462" />
  <package id="System.Buffers" version="4.5.1" targetFramework="net462" />
  <package id="System.Diagnostics.DiagnosticSource" version="4.6.0" targetFramework="net462" />
  <package id="System.Memory" version="4.5.4" targetFramework="net462" />
  <package id="System.Memory.Data" version="1.0.2" targetFramework="net462" />
  <package id="System.Numerics.Vectors" version="4.5.0" targetFramework="net462" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.6.0" targetFramework="net462" />
  <package id="System.Text.Encodings.Web" version="4.7.2" targetFramework="net462" />
  <package id="System.Text.Json" version="4.6.0" targetFramework="net462" />
  <package id="System.Threading.Tasks.Extensions" version="4.5.2" targetFramework="net462" />
  <package id="System.ValueTuple" version="4.5.0" targetFramework="net462" />
```

File: soft-delete/soft-delete.csproj [hunk 10]
```diff
@@ -52,18 +60,46 @@
...
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.WindowsAzure.Storage, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\packages\WindowsAzure.Storage.9.0.0\lib\net45\Microsoft.WindowsAzure.Storage.dll</HintPath>
-    </Reference>
     <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\packages\Newtonsoft.Json.11.0.1-beta1\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
```
Comment (christothes) [inline soft-delete/soft-delete.csproj:51]: Let's clean out all the references here that are not 100% needed.

**Review by christothes — COMMENTED** [review]
Please ensure this builds locally.

File: soft-delete/App.config (hunk 8 above)
Comment (zzkaii) [reply soft-delete/App.config]: Done.

File: soft-delete/packages.config (hunk 9 above)
Comment (zzkaii) [reply soft-delete/packages.config]: Done.

File: soft-delete/soft-delete.csproj (hunk 10 above)
Comment (zzkaii) [reply soft-delete/soft-delete.csproj:51]: Done.

**Comment by v-xuto** [comment]
@christothes We have updated by your comments, please re-review this PR.

**Review by christothes — APPROVED** [review]
